### ISSUE TITLE
Project referencing Shouldly NuGet package is rebuilt every time, because no .pdb file is generated

### ISSUE THREAD

**Issue by loop-evgeny** [shouldly/shouldly#723]
I created a trival C# project referencing Shouldly, like this:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <OutputType>Library</OutputType>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.8.0" />
    <PackageReference Include="NUnit" Version="3.12.0" />
    <PackageReference Include="NUnit3TestAdapter" Version="3.17.0" />
    <PackageReference Include="Shouldly" Version="4.0.1" />
  </ItemGroup>

</Project>
```

`dotnet build` and Visual Studio build rebuild this project every time, even though no changes have been made. VS logs this:

> FastUpToDate: Output 'ShoudlyTest\bin\Debug\netcoreapp3.1\ShoudlyTest.pdb' does not exist, not up to date. (ShoudlyTest)

Indeed, that .pdb file is not generated by the build, even though the default settings are to generate a .pdb file. It still happens even if I try to explicitly set the project to generate a .pdb file:

```
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <DebugType>full</DebugType>
    <DebugSymbols>true</DebugSymbols>
  </PropertyGroup>
```

I see that build/Shouldly.targets in the NuGet package sets `<DebugType>embedded</DebugType>`, which is probably what causes this.

Even without the rebuild issue, Shouldly should not be changing the debug type settings for my project - I might need that .pdb file for other reasons.

**Comment by loop-evgeny** [comment]
Other projects in the solution (not referencing the test project) seem to sometimes be rebuilt now as well, but not every time:

> 1>FastUpToDate: Input 'Other.csproj.nuget.g.props' (25/11/2020 4:49:32 PM) has been modified since the last up-to-date check (25/11/2020 4:49:31 PM), not up to date.

I haven't investigated this further.

**Comment by dashenxian** [comment]
I'm having the same problem，
create a new .net core3.1 ClassLibrary,then referencing Shouldly(4.0.1) on NuGet,build the project with visual studio2019

**Comment by skrysmanski** [comment]
I'm running into the same problem - but with .NET 5.

**Comment by skrysmanski** [comment]
I did some more digging and can confirm that the problem is that Shouldly is setting the debug type to "embedded".

You can confirm this by removing the line `<DebugType>embedded</DebugType>` from the `Shouldly.targets` file. On Windows, this file is located here:

    %USERPROFILE%\.nuget\packages\shouldly\4.0.1\build\Shouldly.targets

It appears that Visual Studio (or MSBuild or whatever) does not expect that the debug symbols type is changed by a NuGet package. Also, since [the file in this repository](https://github.com/shouldly/shouldly/commits/master/src/Shouldly/build.targets) has not changed in the last couple of months I'm guessing the problem is caused by a recent Visual Studio update (most likely 16.8).

As a workaround (for now) you can manually set the debug type in your test projects to "embedded".

**Comment by KirillOsenkov** [comment]
I think it’s a bug in the fast up-to-date check logic in the new project system. Would you mind opening an issue at https://github.com/dotnet/project-system/issues/new and linking to this issue? I think there is a way to add the phantom pdb file into an exclusion list so it’s not considered output, but I can’t remember offhand how to do it.

**Comment by KirillOsenkov** [comment]
https://github.com/dotnet/project-system/blob/master/docs/up-to-date-check.md

@drewnoakes

**Comment by loop-evgeny** [comment]
This is not a problem with the fast up-to-date check, because it also happens with `dotnet build` (ie. without VS involved at all). If you run `dotnet build /v:n` you can see that it runs the `CoreCompile` target:

>          C:\Program Files\dotnet\dotnet.exe exec "C:\Program Files\dotnet\sdk\5.0.101\Roslyn\bincore\csc.dll" ... (lots of args) ...

but if you run `dotnet build /v:n` twice for a project not referencing Shouldly the second time it says:

>        CoreCompile:
>       Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.

**Comment by SimonCropp** [comment]
can u try 4.0.2

**Comment by loop-evgeny** [comment]
v4.0.2 fixes this issue, thanks, but it adds a new one: https://github.com/shouldly/shouldly/issues/738

**Comment by dashenxian** [comment]
It's ok now,[but Empty files are shown as included in project in v4.0.2 #738](https://github.com/shouldly/shouldly/issues/738)

**Comment by greengumby** [comment]
This never was a problem with .net 4.X using  `nuget pack` any ideas why `dotnet pack` requires the PDB file?
